lxd/instance/drivers/driver_qemu.go

```diff
@@ -1763,6 +1773,102 @@ func (d *qemu) deviceStart(deviceName string, rawConfig deviceConfig.Device, ins
 	return runConf, nil
 }
 
+func (d *qemu) deviceAttachBlockDevice(deviceName string, configCopy map[string]string, mounts []deviceConfig.MountEntryItem) error {
+	revert := revert.New()
+	defer revert.Fail()
+
+	// Check if the agent is running.
+	monitor, err := qmp.Connect(d.monitorPath(), qemuSerialChardevName, d.getMonitorEventHandler())
+	if err != nil {
+		return fmt.Errorf("Failed to connect to QMP monitor: %w", err)
+	}
+
+	fields := strings.Split(mounts[0].DevPath, ":")
```

Comment:
A comment here explaining what you're splitting up would be useful to the reader.